### PR TITLE
chore(main/bionic-host): fix downloading ubuntu packages, disable undefined symbols checking, disable `repo` SHA256 verification

### PR DESCRIPTION
**Pull request by twaik**
Replaces #21095

### CODE REVIEW

**Comment by twaik** [comment]
@Grimler91 ping because you authored #21095.

**Comment by twaik** [comment]
Last run is successful (https://github.com/termux/termux-packages/actions/runs/11355611298), I commited revbump to avoid problems after squashing and merging to master.

**Comment by robertkirkman** [comment]
Thank you! I was also building this and when I apply your PR it builds successfully.

> jammy is last known distro which contains `libncurses.so.5` in packages

I'm not sure if maybe you meant the last known Ubuntu distro, but technically, Debian 12 released at a date later than Ubuntu 22.04 and it contains that package.

That means that when I apply this change to your PR it still builds, but maybe there's a specific reason it has to be the Ubuntu 22.04 version that I don't know about.

One possible comment in favor of this suggestion I could make is that, in my experience, https://cdn-fastly.deb.debian.org/ is usually a significantly more reliable mirror than alternatives. However maybe that does not matter at all anymore since you have implemented a loop to retry on failure.

```diff
--- a/packages/bionic-host/build.sh
+++ b/packages/bionic-host/build.sh
@@ -11,8 +11,8 @@ TERMUX_PKG_SKIP_SRC_EXTRACT=true
 
 # Function to obtain the .deb URL
 obtain_deb_url() {
-	# jammy is last known distro which contains `libncurses.so.5` in packages
-	local url="https://packages.ubuntu.com/jammy/amd64/$1/download"
+	# bookworm is last known distro which contains `libncurses.so.5` in packages
+	local url="https://packages.debian.org/bookworm/amd64/$1/download"
 	local retries=5
 	local wait=5
 	local attempt
@@ -22,7 +22,8 @@ obtain_deb_url() {
 		local PAGE="$(curl -s "$url")"
 		>&2 echo page
 		>&2 echo "$PAGE"
-		if deb_url=$(echo "$PAGE" | grep -Eo 'http://.*\.deb' | head -n 1); then
+		if deb_url=$(echo "$PAGE" | grep -Eo 'http://http.us.debian.org/.*amd64.deb' | head -n 1 | \
+		sed -e 's/http:\/\/http.us/https:\/\/cdn-fastly.deb/'); then
 			if [[ -n "$deb_url" ]]; then
 				echo "$deb_url"
 				return 0
@@ -65,7 +66,7 @@ termux_step_get_source() {
 
 		mkdir -p ${TERMUX_PKG_TMPDIR}/${URL##*/}
 		ar x ${TERMUX_PKG_CACHEDIR}/${URL##*/} --output=${TERMUX_PKG_TMPDIR}/${URL##*/}
-		tar xf ${TERMUX_PKG_TMPDIR}/${URL##*/}/data.tar.zst -C ${TERMUX_PKG_SRCDIR}/prefix
+		tar xf ${TERMUX_PKG_TMPDIR}/${URL##*/}/data.tar.xz -C ${TERMUX_PKG_SRCDIR}/prefix
 	done
 
 	termux_download https://storage.googleapis.com/git-repo-downloads/repo ${TERMUX_PKG_CACHEDIR}/repo SKIP_CHECKSUM
```

**Comment by twaik** [comment]
> Debian 12 released at a date later than Ubuntu 22.04 and it contains that package.

Github CI uses ubuntu imqges so I prefer to use Ubuntu's repos for this purpose, but good catch. Thank you.

**Review by truboxl — CHANGES_REQUESTED** [review]

File: packages/bionic-host/build.sh [hunk 1]
```diff
@@ -78,3 +101,16 @@ termux_step_make_install() {
 	cp ${TERMUX_PKG_SRCDIR}/external/icu/icu4c/source/stubdata/icudt58l.dat ${TERMUX_PREFIX}/opt/bionic-host/usr/icu/
 	cp -r ${TERMUX_PKG_SRCDIR}/out/target/product/generic*/system/* ${TERMUX_PREFIX}/opt/bionic-host/
 }
+
+termux_step_massage() {
```
[the remark this answers is not in the document]
Comment (twaik) [reply packages/bionic-host/build.sh]: Done

**Review by truboxl — APPROVED** [review]

**Comment by twaik** [comment]
> Something about needing to deal with Google's source management just feels icky to me.

I can do nothing with it, it is the way AOSP ROM is being built. Sorry.

**Pull request merged**